### PR TITLE
Implement embed_migration! using macros 1.1

### PR DESCRIPTION
**Pull request by weiznich**
* port embed_migration to macros 1.1
* remove codegen_old
* some minor fixes

This is a breaking change, because the root path passed to `embed_migrations!` have changed. It's now CARGO_MANIFEST_HOME, but was before the location of the file that called that macro.

### CODE REVIEW

**Comment by weiznich** [comment]
Travis is failing while building compilertest. Any idea why?

**Comment by killercup** [comment]
Sorry, I haven't had time to look at this so far :(

The CI error is because this compiles compiletest 0.2.4 which expects a nightly from 2026-10-07 (see <https://github.com/laumann/compiletest-rs/pull/50>; we use 2016-09-29). You probably need to change the dependency to `=0.2.3`.

**Comment by weiznich** [comment]
Using compiletest 0.2.3 didn't fix this.

File: diesel_compile_tests/Cargo.toml [hunk 1]
```diff
@@ -6,4 +6,4 @@ authors = ["Sean Griffin <sean@seantheprogrammer.com>"]
 [dependencies]
 diesel = { version = "0.7.1", features = ["sqlite", "postgres"] }
 diesel_codegen = { version = "0.7.2" }
-compiletest_rs = "0.2.1"
+compiletest_rs = "0.2.3"
```
Comment (weiznich) [inline diesel_compile_tests/Cargo.toml]: Thanks, didn't know about this one.

**Comment by sgrif** [comment]
Can you submit the API change as a separate PR without re-implementing it? That will make it much easier to review.

**Comment by sgrif** [comment]
I'm wanting to release today, so in order to make sure things move quickly I'm going to go ahead and take over here and make the changes I'd like to see (and split up the PR into smaller commits). I'll make sure you're credited on the commits that go in.

**Comment by sgrif** [comment]
Relevant code pulled into #471 and #472.

**Pull request closed without merging**